### PR TITLE
Analytics: Simultaneous video recording

### PR DESCRIPTION
**Pull request by hobinjk-ptc**
With the latest ios build and the corresponding core-addon PR this should enable the ability to click "Show Spatial Video" while viewing a highlight animation playing to see the corresponding video at that time. Note that there's no handling of the recording device's orientation, suggestions welcome.

### CODE REVIEW

**Comment by hobinjk-ptc** [comment]
As per the other PR, this now also adds labels to analytics sessions

File: src/humanPose/Animation.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        // Adjusted cursorTime that might be tracking a
+        // wobbly-relative-to-real-time video playback
+        let cursorTimeAdj = this.cursorTime;
+        if (this.isVideoPlayerInControl(cursorTimeAdj) && this.playing) {
+            if (this.animationState !== AnimationState.video) {
+                this.startVideoPlayback(cursorTimeAdj);
+            } else {
+                cursorTimeAdj = this.cursorTimeFromVideo();
```
Comment (ptc-mklaudiny) [inline src/humanPose/Animation.js:92]: So if I understand correctly, video player does equivalent of 'this.cursorTime += dt;' and the calculation here is overriden.

**Comment by ptc-mklaudiny** [comment]
It seems to me that the last piece for perfect sync between human poses and video frames is missing. Vuforia timestamp for a pose is stored in lastUpdateDataTS in HumanObject. 
https://github.com/ptcrealitylab/vuforia-spatial-edge-server/blob/4c2a3f3797b304c22ab715d54493743e444bcdb9/server.js#L3604-L3605 
This timestamp should be used for Pose instances when replaying them from recorded data.

**Comment by hobinjk-ptc** [comment]
In src/humanPose/index.js I do swap to lastUpdateDataTS https://github.com/ptcrealitylab/vuforia-spatial-toolbox-userinterface/blob/main/src/humanPose/index.js#L256 though it's possible that it's getting overwritten somewhere

File: src/humanPose/Animation.js (hunk 1 above)
Comment (hobinjk-ptc) [reply src/humanPose/Animation.js:92]: This Animation does `this.cursorTime += dt`. The VideoPlayer uses its color video element's currentTime property which exactly tracks where the video is.  This code `this.cursorTime` with the more-exact video-derived cursorTimeFromVideo() for pose display. It doesn't write to `this.cursorTime` to guarantee that the animation continues eventually. For example, if the video starts buffering and pauses, then `this.cursorTimeFromVideo()` will stop moving forwards as well. Not writing guarantees that the loop will eventually start from the beginning again

**Comment by ptc-mklaudiny** [comment]
I somehow overlooked that. All good.

**Pull request merged**